ckanext/security/validators.py

```diff
@@ -5,13 +5,16 @@
 from ckan import authz
 from ckan.common import _
 from ckan.lib.navl.dictization_functions import Missing, Invalid
+# (canada fork only): more configs
+from ckan.plugins.toolkit import config, asbool
 
-MIN_PASSWORD_LENGTH = 8
-MIN_LEN_ERROR = (
-    'Your password must be {} characters or longer, and consist of at least '
-    'three of the following character sets: uppercase characters, lowercase '
-    'characters, digits, punctuation & special characters.'
+MIN_PASSWORD_LENGTH = int(config.get('ckanext.security.min_password_length', 8))
+MIN_LEN_ERROR = 'Your password must be {} characters or longer.'
+COMPLEXITY_ERROR = (
+    'Your password must consist of at least three of the following character sets: '
+    'uppercase characters, lowercase characters, digits, punctuation & special characters.'
 )
+NZISM_COMPLIANT = asbool(config.get('ckanext.security.nzism_compliant_passwords', True))
```

Comment:
loading config at module level means we can't write tests that exercise different values for this option in the normal way

Reply:
Realllly. okay that is good to know. I will just move the definitions inside the validator method then

Reply:
yeah, nothing's lazy in python. If it's at the module level then it's executed only once at import time.